configure

```diff
@@ -142,15 +142,15 @@ includedir=${includedir:-$prefix/include}
 mandir=${mandir:-$prefix/share/man}
 
 # define CFLAGS and LDFLAGS if no environment variables defined
-if test -z ${CFLAGS}; then
-  CFLAGS='-Wall -Werror -std=gnu99 -pedantic -fstack-protector -fPIE -fPIC -fno-strict-overflow -fno-delete-null-pointer-checks -fwrapv'
+if test -z "${CFLAGS}"; then
+  CFLAGS="-Wall -Werror -std=gnu99 -pedantic -fstack-protector-strong -fPIE -fPIC -fno-strict-overflow -fno-delete-null-pointer-checks -fwrapv"
```

Comment:
The if-else-branch with repeated flags looks error-prone. Could you consider doing something like this:

    # flags we always want to set
    NEW_CFLAGS="-Wall -Werror -std=gnu99 -pedantic -fstack-protector-strong -fPIE -fPIC"

    # flags we set if no external CFLAGS is set
    IF_NO_CFLAGS="-fno-strict-overflow -fno-delete-null-pointer-checks -fwrapv"

    CFLAGS="${CFLAGS:-$IF_NO_CFLAGS} ${NEW_CFLAGS}"

(I'm not the repo owner.)

Reply:
i would like to limit this PR to adding -fstack-protector-strong only. surely the repo owner can accept your fix too or you can open a separate PR for this.

i have a concern though, it looks like the `${CFLAGS:-$IF_NO_CFLAGS}` is a bashism. are we sure the `configure` script is run in `bash` only?

Reply:
`#! /bin/bash` at the top of the file indicates that.